src/main/kotlin/network/warzone/mars/player/achievements/AchievementEmitter.kt

```diff
@@ -0,0 +1,55 @@
+package network.warzone.mars.player.achievements
+
+import network.warzone.api.database.models.Achievement
+import network.warzone.mars.Mars
+import network.warzone.mars.api.ApiClient
+import network.warzone.mars.api.socket.OutboundEvent
+import network.warzone.mars.api.socket.models.PlayerAchievementData
+import network.warzone.mars.api.socket.models.SimplePlayer
+import network.warzone.mars.player.feature.PlayerFeature
+import network.warzone.mars.player.models.PlayerProfile
+import org.bukkit.Bukkit
+import org.bukkit.ChatColor
+import org.bukkit.entity.Player
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+
+// A class for adding completed achievements to a player's profile.
+class AchievementEmitter(private val achievement: Achievement) {
+    // Fetch a player profile as needed.
+    fun emit(player: Player) {
+        Mars.async {
+            val profile = PlayerFeature.fetch(player.name)!!
+            emit(profile)
+        }
+    }
+
+    // Add an achievement to the specified profile.
+    fun emit(profile: PlayerProfile) {
+        if (profile.stats.achievements.containsKey(achievement._id.toString())) return // Player already has achievement.
+
+        // Print achievement earn to player and console.
+        val player = Bukkit.getPlayer(profile._id)
+        if (player != null) {
+            player.sendMessage("${ChatColor.GRAY}You've earned an achievement: ${ChatColor.AQUA}" + achievement.name + "${ChatColor.GRAY}!")
+            println("Achievement " + achievement.name + " earned by " + player.name)
+        }
+
+        // Emit achievement completion to the database.
+        ApiClient.emit(
+            OutboundEvent.PlayerAchievement, PlayerAchievementData(
+                SimplePlayer(profile._id, profile.name),
+                achievement._id,
+                Date().time
+            )
+        )
+    }
+
+    private fun getDate(): String {
```

Comment:
this seems a bit generic to be in an achievement-specific file, maybe consider moving it to a util file?

Reply:
This is actually an unused method now that I'm seeing it again. I'll simply remove it from the code.